libcontainerd/process_linux.go

```diff
@@ -79,7 +79,9 @@ func (r emptyReader) Read(b []byte) (int, error) {
 
 func openReaderFromFifo(fn string) io.Reader {
 	r, w := io.Pipe()
+	c := make(chan struct{})
 	go func() {
+		close(c)
 		stdoutf, err := os.OpenFile(fn, syscall.O_RDONLY, 0)
```

Comment:
This `OpenFile` will block until the other side opens the same fifo. If we block here then the process is never started. I think `close()` shouldn't cause the goroutine to get unscheduled. As I wrote there are probably better fixes, but I'm not sure we are comfortable making the other ones for `v1.11.1`.